osquery/filesystem/linux/proc.h

```diff
@@ -0,0 +1,277 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#pragma once
+
+#include <unordered_map>
+
+#include <arpa/inet.h>
+#include <linux/limits.h>
+#include <unistd.h>
+
+#include <boost/filesystem.hpp>
+
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+
+#include "osquery/core/conversions.h"
+
+namespace osquery {
+extern const char* kLinuxProcPath;
+
+struct ProcessSocket final {
```

Comment:
Can you initialize the `int`s and `uint16_t`s? No constructor needed, just `{0}`.